go/arrow/ipc/cmd/arrow-json-integration-test/main_test.go

```diff
@@ -39,8 +39,8 @@ func TestIntegration(t *testing.T) {
 			if err != nil {
 				t.Fatal(err)
 			}
+			defer os.Remove(af1.Name())
 			defer af1.Close()
-			defer os.RemoveAll(af1.Name())
```

Comment:
same comment than above: perhaps replace with a per-sub-test top-level tmp dir that is being `defer os.RemoveAll(tmpdir)` ?